routers/web/auth/auth.go

```diff
@@ -211,6 +213,11 @@ func SignInPost(ctx *context.Context) {
 			log.Info("Failed authentication attempt for %s from %s: %v", form.UserName, ctx.RemoteAddr(), err)
 			ctx.Data["Title"] = ctx.Tr("auth.prohibit_login")
 			ctx.HTML(http.StatusOK, "user/auth/prohibit_login")
+		} else if errors.As(err, &textprotoErr) {
```

Comment:
Pointer of Pointer?

Reply:
Yes, quite surprise to me too!  
ref: https://pkg.go.dev/errors#As

Reply:
I know using errors.As  is somewhat unintuitive, But `errors.As` has been used in other places of Gitea like:

```
modules/git/command.go
modules/git/commit.go
routers/web/repo/release.go
routers/api/packages/container/container.go
```
So I assume using errors.As is allowed.

Anyway, here it is necessary to extract textproto.Error so that the detailed error code and msg can be logged when authentication fails. we are using aliyun mail, the error logged is like this:
```
...ers/web/auth/auth.go:218:SignInPost() [I] Failed authentication attempt for xxxx@xxx.com from 127.0.0.1:39512: (code: 526, msg: Authentication failure[0])
```
I search the error code 526, but it seems that the code may vary with different email vendor. When `textproto.Error` occurs  it does not necessarily mean that it is an authentication error, it is better to catch and log the error so that there is a way that Admin can go through the log and find the actual reason when authentication fails.